src/backbone_router/smcroute_manager.cpp

```diff
@@ -0,0 +1,193 @@
+/*
+ *    Copyright (c) 2020, The OpenThread Authors.
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *    1. Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *    3. Neither the name of the copyright holder nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *    ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   The file implements the SMCRoute manager.
+ */
+
+#include "backbone_router/smcroute_manager.hpp"
+
+#include <assert.h>
+#include <unistd.h>
+
+#include <openthread/backbone_router_ftd.h>
+
+#include "common/code_utils.hpp"
+#include "utils/system_utils.hpp"
+
+namespace otbr {
+
+namespace BackboneRouter {
+
+void SMCRouteManager::Init(const std::string &aThreadIfName, const std::string &aBackboneIfName)
+{
+    assert(!mEnabled);
+
+    mThreadIfName   = aThreadIfName;
+    mBackboneIfName = aBackboneIfName;
+
+    StartSMCRouteService();
+}
+
+void SMCRouteManager::Enable(void)
+{
+    otbrError error = OTBR_ERROR_NONE;
+
+    VerifyOrExit(!mEnabled);
+
+    mEnabled = true;
+
+    Flush();
+
+    // Add mroute rules: 65520 (0xfff0) allow outbound for MA scope >= admin (4)
+    SuccessOrExit(error = AllowOutboundMulticast());
+
+    // Add mroute rules for the current Multicast Listeners Table
+    for (const Ip6Address &address : mListenerSet)
+    {
+        SuccessOrExit(error = AddRoute(address));
+    }
+
+exit:
+    otbrLogResult(error, "SMCRouteManager::Start");
```

Comment:
nit: better to match the method name?

```suggestion
    otbrLogResult(error, "SMCRouteManager::Enable");
```

 may consider using `__FUNCTION__` instead.